### ISSUE TITLE
Register version information on Windows

### ISSUE THREAD

**Issue by lunacd** [rust-lang/rustup#3326]
### Problem you are trying to solve

Without the version information registered, `winget` is difficult to work with:

```
Rustup: the Rust toolchain installer      Rustlang.Rustup                           Unknown           1.25.2    winget
```

### Solution you'd like

Rustup installer ideally should register its version with Windows so that this does not happen.

**Comment by vilgotf** [comment]
Duplicate of #3102. Fix is available on `master` and will be included in rustup 1.26

**Comment by 0xPoe** [comment]
Duplicate of https://github.com/rust-lang/rustup/issues/3102

Thanks!
